electrum_nmc/electrum/commands.py

```diff
@@ -1879,9 +1909,21 @@ async def add_request(self, amount, memo='', expiration=3600, force=False, walle
                 addr = wallet.create_new_address(False)
             else:
                 return False
+        commitment = None
+        if identifier:
+            identifier= name_from_str(identifier, Encoding.ASCII)
+            identifier_hex = name_to_str(identifier, Encoding.HEX)
+            identifier_formatted = format_name_identifier(identifier)
+            name_new_result = await self.name_new(identifier=identifier_hex, 
+                                                name_encoding=Encoding.HEX,
+                                                destination=addr,
+                                                unsigned=True, 
+                                                commitment_only=True)
+            commitment = name_new_result["commitment"]
+            memo = memo if memo else  f"Pre-registration: {identifier_formatted}"
```

Comment:
Don't include the identifier in the memo, because that leaks into the URI and would allow frontrunning. (Also I *think* this doesn't even work because `memo` default to an empty string rather than `None`?)

Reply:
If it's empty is when we set it, the user can also pass the memo

Reply:
Removed this